File: spring-rest-configuration-autodiscover.scala/src/main/kotlin/demo/service/core/PropertyStringService.kt
package demo.service.core

abstract class PropertyStringService : AbstractPropertyService<String>() {

    override fun getType() = Type.STRING

    public override fun fromString(input: String?): String =
            input ?: getDefaultValue()

    public override fun toString(input: String?): String =
            input ?: getDefaultValue()
}

File: spring-rest-configuration-autodiscover.scala/src/main/scala/demo/service/core/PropertyStringService.scala
package demo.service.core

abstract class PropertyStringService extends AbstractPropertyService[String] {

  override def getType: Type.Value = Type.STRING

  override def fromString(input: String): String =
    if (input == null) getDefaultValue else input

  override def toString(input: String): String =
    if (input == null) getDefaultValue else input
}
